fabric-structure-api-v1/src/main/java/net/fabricmc/fabric/api/structure/v1/StructurePoolAddCallback.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.structure.v1;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * A callback for newly added structure pools.
+ *
+ * <p><strong>Word of warning</strong>: Mods may be editing on the structure pool from user configured resource pack
+ * than what's from the builtin minecraft resources.
+ *
+ * <p>Example usage:
+ * <pre>{@code
+ * StructurePoolAddCallback.EVENT.register(structurePool -> {
+ * 	if (structurePool.getStructurePool().getId().getPath().equals("village/common/butcher_animals")) {
```

Comment:
```suggestion
 * 	if (structurePool.getUnderlying().getId().getPath().equals("village/common/butcher_animals")) {
```
(Minor thing) I'd also prefer checking the whole ID (it's possible that someone has a modded structure pool named `their_namespace:village/common/butcher_animals`.

Reply:
Also this example should probably include some pseudocode that only registers this callback if a config entry is enabled (as otherwise users cannot turn off pool modifications from this mod on user pools easily)